packages/tables/src/Table/Concerns/CanGroupRecords.php

```diff
@@ -82,6 +86,12 @@ public function defaultGroup(string | Group | null $group): static
      */
     public function groups(array | Closure $groups): static
     {
+        foreach ($groups as $group) {
+            if ($group instanceof Group) {
+                $group->table($this);
+            }
+        }
```

Comment:
If groups are passed in a closure I don't think the table gets passed, I think the table needs to be injected in when the groups property is consumed